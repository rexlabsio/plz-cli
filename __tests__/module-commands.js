const pify = require('pify');
const path = require('path');
const fs = pify(require('fs'));
const fse = require('fs-extra');
const { snapshotStdout, CLI_PATH, CLI_PROJECT_PATH } = require('./.utils');

const TEST_OUTPUT_PATH = path.resolve(CLI_PROJECT_PATH, '.test-output');
const PROJECT_NAME = 'test-module';
const PROJECT_PATH = path.resolve(TEST_OUTPUT_PATH, PROJECT_NAME);

describe('module (node)', () => {
  const DEFAULT_TIMEOUT_INTERVAL = jasmine.DEFAULT_TIMEOUT_INTERVAL;
  const NEW_TIMEOUT_INTERVAL = 1000 * 120; // 120 seconds

  beforeAll(async () => {
    jasmine.DEFAULT_TIMEOUT_INTERVAL = NEW_TIMEOUT_INTERVAL;
    await expect(fse.remove(PROJECT_PATH)).resolves.toBeUndefined();
    await expect(fse.ensureDir(PROJECT_PATH)).resolves.toBeTruthy();
  });
  afterAll(async () => {
    jasmine.DEFAULT_TIMEOUT_INTERVAL = DEFAULT_TIMEOUT_INTERVAL;
    await expect(fse.remove(PROJECT_PATH)).resolves.toBeUndefined();
  });

  it('should create project', async () => {
    await snapshotStdout(CLI_PATH, [
      'create',
      PROJECT_NAME,
      TEST_OUTPUT_PATH,
      '--type',
      'module'
    ]);
    await expect(fs.readdir(PROJECT_PATH)).resolves.toMatchSnapshot();
  });

  it('should install deps in project', async () => {
    await snapshotStdout(
      'yarn',
      ['install', '--ignore-scripts', '--no-progress'],
      { cwd: PROJECT_PATH }
    );
    await expect(
      fs.readdir(path.resolve(PROJECT_PATH, './node_modules'))
    ).resolves.toBeTruthy();
  });

  it('should link current cli into project', async () => {
    await snapshotStdout('yarn', ['unlink'], { cwd: CLI_PROJECT_PATH });
    await snapshotStdout('yarn', ['link'], { cwd: CLI_PROJECT_PATH });
    await snapshotStdout('yarn', ['link', '@rexlabs/plz-cli'], {
      cwd: PROJECT_PATH
    });
  });

  it('should run valid test in project', async () => {
    await snapshotStdout('npm', ['run', 'test'], { cwd: PROJECT_PATH });
  });

  it('should build project', async () => {
    await snapshotStdout('npm', ['run', 'build'], { cwd: PROJECT_PATH });
    await expect(
      fs.readdir(path.resolve(PROJECT_PATH, 'lib'))
    ).resolves.toMatchSnapshot();
  });

  it('should clean project', async () => {
    await snapshotStdout('npm', ['run', 'clean'], { cwd: PROJECT_PATH });
    await expect(
      fs.readdir(path.resolve(PROJECT_PATH, 'lib'))
    ).rejects.toBeDefined();
  });

  // NOTE lochlan: I usually have the cli linked anyway, so I leave it linked.
  //               Sorry if this breaks something on your system.

  // it('should unlink current heidi-cli from machine', async () => {
  //   await snapshotStdout('yarn', ['unlink'], { cwd: CLI_PROJECT_PATH });
  // });
});
