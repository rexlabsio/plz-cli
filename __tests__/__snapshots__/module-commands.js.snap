// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`module (node) should build project 1`] = `"cmd: npm run,build {\\"cwd\\":\\"<CLI_DIR>/.test-output/test-module\\"}"`;

exports[`module (node) should build project 2`] = `
"stdoutDump:

> @rexlabs/test-module@1.0.0 build <CLI_DIR>/.test-output/test-module
> plz build


 -:  p l z   b u i l d  :-

"
`;

exports[`module (node) should build project 3`] = `
"stderrDump:
✔ Built bundles.
"
`;

exports[`module (node) should build project 4`] = `
Array [
  "index.js",
]
`;

exports[`module (node) should clean project 1`] = `"cmd: npm run,clean {\\"cwd\\":\\"<CLI_DIR>/.test-output/test-module\\"}"`;

exports[`module (node) should clean project 2`] = `
"stdoutDump:

> @rexlabs/test-module@1.0.0 clean <CLI_DIR>/.test-output/test-module
> plz clean


 -:  p l z   c l e a n  :-

"
`;

exports[`module (node) should clean project 3`] = `
"stderrDump:
✔ Cleaned.
"
`;

exports[`module (node) should create project 1`] = `"cmd: <CLI_DIR>/index.js create,test-module,<CLI_DIR>/.test-output,--type,module {}"`;

exports[`module (node) should create project 2`] = `
"stdoutDump:

 -:  p l z   c r e a t e  :-

"
`;

exports[`module (node) should create project 3`] = `
"stderrDump:
✔ Fetched scaffold metadata
✔ Generated 'Module': .test-output/test-module
"
`;

exports[`module (node) should create project 4`] = `
Array [
  "package.json",
  "readme.md",
  "src",
]
`;

exports[`module (node) should install deps in project 1`] = `"cmd: yarn install,--ignore-scripts,--no-progress {\\"cwd\\":\\"<CLI_DIR>/.test-output/test-module\\"}"`;

exports[`module (node) should install deps in project 2`] = `
"stdoutDump:
"
`;

exports[`module (node) should install deps in project 3`] = `
"stderrDump:
warning babel-plugin-inferno@1.7.0: All versions before 3.x are deprecated!
warning babel-preset-es2016@6.22.0: We're really excited that you're trying to use ES2016 syntax, but instead of making more yearly presets, Babel has a better preset that we recommend you use instead: npm install babel-preset-env --save-dev. Give us a follow @babeljs to get more updates
<UNMET PEER DEP WARNING>
<UNMET PEER DEP WARNING>
<UNMET PEER DEP WARNING>
<UNMET PEER DEP WARNING>
<UNMET PEER DEP WARNING>
<UNMET PEER DEP WARNING>
<UNMET PEER DEP WARNING>
warning \\"glob-loader@0.3.0\\" has incorrect peer dependency \\"glob@*\\".
warning \\"html-webpack-plugin@2.30.1\\" has incorrect peer dependency \\"webpack@1 || ^2 || ^2.1.0-beta || ^2.2.0-rc || ^3\\".
warning \\"html-webpack-plugin@2.28.0\\" has incorrect peer dependency \\"webpack@1 || ^2 || ^2.1.0-beta || ^2.2.0-rc\\".
<UNMET PEER DEP WARNING>
<UNMET PEER DEP WARNING>
warning \\"storybook-readme@3.0.6\\" has incorrect peer dependency \\"@storybook/addons@^3.0.0-alpha.0\\".
<UNMET PEER DEP WARNING>
<UNMET PEER DEP WARNING>
warning \\"stats-webpack-plugin@0.6.1\\" has incorrect peer dependency \\"webpack@^1.0||^2.1.0-beta||^2.2.0-rc\\".
<UNMET PEER DEP WARNING>
<UNMET PEER DEP WARNING>
<UNMET PEER DEP WARNING>
<UNMET PEER DEP WARNING>
<UNMET PEER DEP WARNING>
<UNMET PEER DEP WARNING>
warning \\"@storybook/addon-storyshots@3.2.8\\" has incorrect peer dependency \\"@storybook/addons@^3.2.6\\".
warning \\"@storybook/addon-storyshots@3.2.8\\" has incorrect peer dependency \\"@storybook/channels@^3.2.0\\".
warning \\"@storybook/addon-storyshots@3.2.8\\" has incorrect peer dependency \\"@storybook/react@^3.2.8\\".
warning \\"@storybook/addon-storyshots@3.2.8\\" has incorrect peer dependency \\"babel-core@^6.25.0\\".
<UNMET PEER DEP WARNING>
warning \\"@storybook/addon-storyshots@3.2.8\\" has incorrect peer dependency \\"react-test-renderer@*\\".
warning \\"babel-loader@7.1.2\\" has incorrect peer dependency \\"babel-core@6 || 7 || ^7.0.0-alpha || ^7.0.0-beta || ^7.0.0-rc\\".
warning \\"babel-loader@7.1.2\\" has incorrect peer dependency \\"webpack@2 || 3\\".
warning \\"babel-loader@6.2.10\\" has incorrect peer dependency \\"babel-core@^6.0.0\\".
warning \\"babel-loader@6.2.10\\" has incorrect peer dependency \\"webpack@1 || 2 || ^2.1.0-beta || ^2.2.0-rc\\".
warning \\"babel-preset-react-app@3.0.2\\" has incorrect peer dependency \\"babel-runtime@^6.23.0\\".
warning \\"glamorous@4.8.0\\" has incorrect peer dependency \\"glamor@>=2\\".
<UNMET PEER DEP WARNING>
<UNMET PEER DEP WARNING>
<UNMET PEER DEP WARNING>
warning \\"url-loader@0.5.9\\" has incorrect peer dependency \\"file-loader@*\\".
warning \\"url-loader@0.5.7\\" has incorrect peer dependency \\"file-loader@*\\".
warning \\"webpack-dev-middleware@1.12.0\\" has incorrect peer dependency \\"webpack@^1.0.0 || ^2.0.0 || ^3.0.0\\".
warning \\"webpack-dev-middleware@1.9.0\\" has incorrect peer dependency \\"webpack@1 || ^2.1.0-beta || ^2.2.0-rc.0\\".
<UNMET PEER DEP WARNING>
<UNMET PEER DEP WARNING>
<UNMET PEER DEP WARNING>
<UNMET PEER DEP WARNING>
warning \\"karma-mocha-reporter@2.2.2\\" has incorrect peer dependency \\"karma@>=0.13\\".
warning \\"karma-phantomjs-launcher@1.0.2\\" has incorrect peer dependency \\"karma@>=0.9\\".
warning \\"karma-webpack@2.0.1\\" has incorrect peer dependency \\"webpack@^1.1.0 || ^2 || ^2.1.0-beta.0 || ^2.2.0-rc.0\\".
warning \\"extract-text-webpack-plugin@2.1.0\\" has incorrect peer dependency \\"webpack@^2.2.0\\".
warning \\"npm-install-webpack2-plugin@4.1.0\\" has incorrect peer dependency \\"webpack@^1.12.0 || ^2.1.0-beta.0 || ^2.1.0\\".
<UNMET PEER DEP WARNING>
<UNMET PEER DEP WARNING>
warning \\"react-datetime@2.10.1\\" has incorrect peer dependency \\"moment@>=2.16.0\\".
<UNMET PEER DEP WARNING>
<UNMET PEER DEP WARNING>
<UNMET PEER DEP WARNING>
warning \\"ajv-keywords@2.1.0\\" has incorrect peer dependency \\"ajv@>=5.0.0\\".
warning \\"ajv-keywords@1.5.1\\" has incorrect peer dependency \\"ajv@>=4.10.0\\".
warning \\"uglifyjs-webpack-plugin@0.4.6\\" has incorrect peer dependency \\"webpack@^1.9 || ^2 || ^2.1.0-beta || ^2.2.0-rc || ^3.0.0\\".
<UNMET PEER DEP WARNING>
<UNMET PEER DEP WARNING>
<UNMET PEER DEP WARNING>
<UNMET PEER DEP WARNING>
<UNMET PEER DEP WARNING>
<UNMET PEER DEP WARNING>
<UNMET PEER DEP WARNING>
<UNMET PEER DEP WARNING>
<UNMET PEER DEP WARNING>
<UNMET PEER DEP WARNING>
warning \\"react-icon-base@2.0.7\\" has incorrect peer dependency \\"prop-types@*\\".
<UNMET PEER DEP WARNING>
<UNMET PEER DEP WARNING>
<UNMET PEER DEP WARNING>
<UNMET PEER DEP WARNING>
<UNMET PEER DEP WARNING>
<UNMET PEER DEP WARNING>
warning Ignored scripts due to flag.
"
`;

exports[`module (node) should link current cli into project 1`] = `"cmd: yarn unlink {\\"cwd\\":\\"<CLI_DIR>\\"}"`;

exports[`module (node) should link current cli into project 2`] = `
"stdoutDump:
"
`;

exports[`module (node) should link current cli into project 3`] = `
"stderrDump:
"
`;

exports[`module (node) should link current cli into project 4`] = `"cmd: yarn link {\\"cwd\\":\\"<CLI_DIR>\\"}"`;

exports[`module (node) should link current cli into project 5`] = `
"stdoutDump:
"
`;

exports[`module (node) should link current cli into project 6`] = `
"stderrDump:
"
`;

exports[`module (node) should link current cli into project 7`] = `"cmd: yarn link,@rexlabs/plz-cli {\\"cwd\\":\\"<CLI_DIR>/.test-output/test-module\\"}"`;

exports[`module (node) should link current cli into project 8`] = `
"stdoutDump:
"
`;

exports[`module (node) should link current cli into project 9`] = `
"stderrDump:
"
`;

exports[`module (node) should run valid test in project 1`] = `"cmd: npm run,test {\\"cwd\\":\\"<CLI_DIR>/.test-output/test-module\\"}"`;

exports[`module (node) should run valid test in project 2`] = `
"stdoutDump:

> @rexlabs/test-module@1.0.0 test <CLI_DIR>/.test-output/test-module
> plz test


 -:  p l z   t e s t  :-

"
`;

exports[`module (node) should run valid test in project 3`] = `
"stderrDump:
fallbackLoader option has been deprecated - replace with \\"fallback\\"
loader option has been deprecated - replace with \\"use\\"
fallbackLoader option has been deprecated - replace with \\"fallback\\"
loader option has been deprecated - replace with \\"use\\"
 PASS  src/__tests__/test-module.js
  testModule
    ✓ should throw an error (<TIMESTAMP>)

Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        <TIMESTAMP>
Ran all test suites.
"
`;
